fastapi_sso/sso/line.py

```diff
@@ -0,0 +1,39 @@
+import warnings
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+import httpx
+from starlette.requests import Request
+
+from fastapi_sso.sso.base import DiscoveryDocument, OpenID, ReusedOauthClientWarning, SSOBase
+
+if TYPE_CHECKING:
+    import httpx
+
+
+class LineSSO(SSOBase):
+    '''Class providing login via Line OAuth'''
+
+    provider = 'line'
+    base_url = 'https://api.line.me/oauth2/v2.1'
+    scope = ['email', 'profile', 'openid']
+
+    async def get_discovery_document(self) -> DiscoveryDocument:
+        '''Get document containing handy urls'''
+        return {
+            'authorization_endpoint': f'https://access.line.me/oauth2/v2.1/authorize',
+            'token_endpoint': f'{self.base_url}/token',
+            'userinfo_endpoint': f'{self.base_url}/userinfo'
+        }
+
+    async def openid_from_response(self, response: dict, session: Optional['httpx.AsyncClient'] = None) -> OpenID:
+        '''Return OpenID from user information provided by Line'''
+        return OpenID(
+            email=response.get('email'),
+            first_name='',
+            last_name='',
```

Comment:
both `first_name` and `last_name` are optional on `OpenID`, I'd prefer to just omit them

Reply:
Also, the linter found some unused imports and stuff, take a look at the pylint job or run `poe lint` locally to see and try to tackle the issues. If you have any troubles, let me know 👍

Reply:
Line didn't provide those information, so I think is better return empty string.

Reply:
Both default to `None`, I feel it's more convenient to check for nones than empty strings, will I change your mind? :smile: https://github.com/tomasvotava/fastapi-sso/blob/master/fastapi_sso/sso/base.py#L49

Reply:
sure, I'll change it to None.